backend/internal/controllers/book.go

```diff
@@ -209,3 +210,62 @@ func (c *BookController) UpdateBookDetails(rw http.ResponseWriter, req *http.Req
 	writeResponse(rw, http.StatusOK, resBookDetails)
 
 }
+
+// increments a book's click count for the current day (/analytics/{id}/inc)
+func (c *BookController) UpdateBookClicks(rw http.ResponseWriter, req *http.Request) {
+	var bookID string = chi.URLParam(req, "id")
+
+	analytics, _ := c.Books.FetchBookAnalytics(req.Context(), bookID, 1)
+
+	if analytics == nil {
+		writeResponse(rw, http.StatusNotFound, "Book analytics not found for requested id")
+		return
+	}
+
+	err := c.Books.UpdateBookAnalytics(req.Context(), bookID)
+
+	if err != nil {
+		writeResponse(rw, http.StatusInternalServerError, "error")
+		return
+	}
+
+	writeResponse(rw, http.StatusNoContent, nil)
+
+}
+
+// fetches daily book clicks in a given range (/analytics/{id}?range=<days>)
+func (c *BookController) GetBookClicks(rw http.ResponseWriter, req *http.Request) {
+	var bookID string = chi.URLParam(req, "id")
+	var dayRange string = req.URL.Query().Get("range")
+	var dRange int
+
+	if dayRange == "" {
+		writeResponse(rw, http.StatusBadRequest, "range missing")
+		return
+	}
+
+	if numDays, err := strconv.Atoi(dayRange); err != nil {
+		writeResponse(rw, http.StatusBadRequest, "could not parse range")
+		return
+	} else if numDays < 1 || numDays > 366 {
+		writeResponse(rw, http.StatusBadRequest, "Range out of bounds")
+		return
+	} else {
+		dRange = numDays
+	}
```

Comment:
You can change `if numDays, err :=` to `if dRange, err :=` in your if statement, and get rid of the `else` block - since dRange is already defined it won't be scoped to just the if statement.

You could also then split the `else if` into a separate `if` statement using dRange instead of numDays

Actually, I would probably just rename dRange to numDays, I think that's a better variable name.